Editor/ReactiveObjects/AnimationGeneration/ReactiveObjectPass.cs

```diff
@@ -26,34 +26,48 @@ internal partial class ReactiveObjectPass
         
         private AnimationClip _initialStateClip;
         private bool _writeDefaults;
-        
+
+        private AnimatorCombiner _combineSession;
+        private AnimatorController _animController;
+
         public ReactiveObjectPass(ndmf.BuildContext context)
         {
             this.context = context;
         }
 
         internal void Execute()
         {
+            var fxController = FindFxController();
+
             // Having a WD OFF layer after WD ON layers can break WD. We match the behavior of the existing states,
             // and if mixed, use WD ON to maximize compatibility.
-            _writeDefaults = MergeAnimatorProcessor.ProbeWriteDefaults(FindFxController().animatorController as AnimatorController) ?? true;
-            
+            _writeDefaults = MergeAnimatorProcessor.ProbeWriteDefaults(fxController.animatorController as AnimatorController) ?? true;
+
+            _combineSession = new AnimatorCombiner(context, fxController.animatorController != null ? fxController.animatorController.name : "FX (Reactive Objects)");
```

Comment:
Recreating the AnimatorCombiner while AnimationServicesContext is active will break the commit actions held in the AnimationDatabase, which holds a series of delegates which write the modified or replaced Motions back to the states they came from: https://github.com/bdunderscore/modular-avatar/blob/main/Editor/Animation/AnimationDatabase.cs#L100

Probably a better short-term approach would be to detect which parameters need this, and force them to be floats during Merge Animator processing. Longer-term I'm working on a redesign of the AnimationServicesContext and AnimatorCombiner to play nicer together.

Reply:
It might be enough to add logic to `ParameterAssignerPass` to generate a Merge Animator component, which merges an animator that declares all relevant parameters (as floats) but has zero layers...